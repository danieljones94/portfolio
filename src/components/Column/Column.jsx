import React, { Component } from "react";
import Col from "react-bootstrap/Col";
import styles from "./Column.module.scss";

class Column extends Component {
  state = { hover: false };
  toggleHover = () => {
    this.setState({ hover: !this.state.hover });
  };

  render() {
    let style;
    this.state.hover
      ? (style = {
          cursor: "pointer",
          transform: "scale(1.1)",
          transition: "all 1s ease",
          color: "white",
          backgroundColor: "#4a738c",
          padding: "3vh"
        })
      : (style = {
          cursor: "pointer",
          color: "transparent",
          padding: "3vh"
        });

    return (
      <div className={styles.columnContainer}>
        <Col
          //   xs={this.props.xs}
          //   md={this.props.md}
          style={style}
          className={this.props.className}
          onMouseEnter={event => this.toggleHover(event)}
          onMouseLeave={event => this.toggleHover(event)}
        />
        <p style={style} className={styles.text}>
          {this.props.text}
        </p>
      </div>
    );
  }
}

export default Column;
