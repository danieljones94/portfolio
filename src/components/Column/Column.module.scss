.columnContainer {
  .text {
    max-width: 20vw;
    position: relative;
    bottom: 22vh;
  }
}
