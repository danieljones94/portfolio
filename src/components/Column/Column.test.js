import Column from "./Column";
import React from "react";
import { mount } from "enzyme";

let mockData =
  "I have  experience working with HTML, CSS, SASS and Javascript, completing various challenges and personal projects";

describe("column tests", () => {
  let testFunction;
  let component;

  beforeEach(() => {
    testFunction = jest.fn();
    component = mount(
      <Column text={mockData} onMouseEnter={() => testFunction()} />
    );
  });

  test("should match the snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  test("should contain a HTML p tag", () => {
    expect(component.find("p").length).toEqual(1);
  });

  test("should receive data that is passed down as props", () => {
    expect(component.props().text).toEqual(mockData);
  });

  test("should render data passed down as props", () => {
    expect(component.text()).toContain(mockData);
  });

  test("should toggle the hover state when the enters the component", () => {
    expect(component.state("hover")).toEqual(false);
    component.instance().toggleHover();
    expect(component.state("hover")).toEqual(true);
  });
});
