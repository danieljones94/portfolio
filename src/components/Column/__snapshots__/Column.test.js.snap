// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`column tests should match the snapshot 1`] = `
<Column
  onMouseEnter={[Function]}
  text="I have  experience working with HTML, CSS, SASS and Javascript, completing various challenges and personal projects"
>
  <div
    className="columnContainer"
  >
    <Col
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "transparent",
          "cursor": "pointer",
          "padding": "3vh",
        }
      }
    >
      <div
        className="col"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "transparent",
            "cursor": "pointer",
            "padding": "3vh",
          }
        }
      />
    </Col>
    <p
      className="text"
      style={
        Object {
          "color": "transparent",
          "cursor": "pointer",
          "padding": "3vh",
        }
      }
    >
      I have  experience working with HTML, CSS, SASS and Javascript, completing various challenges and personal projects
    </p>
  </div>
</Column>
`;
